### PR TITLE
PBDL model for 物理感知神经网络流固耦合计算加速方法研究-王兆坤

### PR DESCRIPTION
**Pull request by cloud2009**
### PR types
New Features

### PR changes
Others. 
Add an example about PBDL model used in '物理感知神经网络流固耦合计算加速方法研究'

### Describe
This is the physics-based deep learning model to predict body force used in immerse boundary method. It can accelerate the traditional LBM calculation by this rapid predicted body force.

### CODE REVIEW

File: examples/PBDL/forceInfo.csv
Comment (HydrogenSulfate) [inline examples/PBDL/forceInfo.csv:1]: 该文件已上传：<https://paddle-org.bj.bcebos.com/paddlescience%2Fdatasets%2FPBDL%2FforceInfo.csv>

File: examples/PBDL/ibm.py [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+
+    def __len__(self):
+        return len(self.data)
+
+    def __getitem__(self, idx):
+        return self.x1[idx], self.x2[idx], self.y[idx]
+
+
+def main(OUTPUT_DIR):
+
```
Comment (HydrogenSulfate) [inline examples/PBDL/ibm.py:61]:
```suggestion
```

File: examples/PBDL/ibm.py [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+            y_batch.to(device),
+        )
+        break
+    for i, (x1_test, x2_test, y_test) in enumerate(test_loader):
+        x1_test, x2_test, y_test = (
+            x1_test.to(device),
+            x2_test.to(device),
+            y_test.to(device),
+        )
+        break
```
Comment (HydrogenSulfate) [inline examples/PBDL/ibm.py:177]: 这两段for循环的作用是？

File: examples/PBDL/ibm.py [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+                        print(f"{epoch + 1}, {L2_error_training:.15f}", file=file1)
+                with paddle.no_grad():
+                    test_prediction = model(x1_test, x2_test)
+                    test_prediction = test_prediction.detach().cpu().numpy()
+                    test_target = y_test.detach().cpu().numpy()
+                    L2_error_testing = np.sqrt(
+                        np.linalg.norm(test_prediction - test_target)
+                        / np.linalg.norm(test_target)
+                    )
+                    with open(f"{OUTPUT_DIR}/TestingLoss_L2.dat", "a") as file2:
```
Comment (HydrogenSulfate) [inline examples/PBDL/ibm.py]: 文件路径请使用os.path.join来拼接

File: examples/PBDL/output/TestingLoss_L2.dat
Comment (HydrogenSulfate) [inline examples/PBDL/output/TestingLoss_L2.dat:1]: 输出的日志不用上传

File: examples/PBDL/output/TrainingLoss_L2.dat
Comment (HydrogenSulfate) [inline examples/PBDL/output/TrainingLoss_L2.dat:1]: 输出的日志不用上传

File: examples/PBDL/output/checkpoint.pth
Comment (HydrogenSulfate) [inline examples/PBDL/output/checkpoint.pth:1]: 请转换成`.pdparams`格式的文件

File: examples/PBDL/output/loss.dat
Comment (HydrogenSulfate) [inline examples/PBDL/output/loss.dat:1]: 输出的日志不用上传

File: examples/PBDL/output/nohup.out
Comment (HydrogenSulfate) [inline examples/PBDL/output/nohup.out:1]: 输出的日志不用上传

File: examples/PBDL/output/trained_model.pth
Comment (HydrogenSulfate) [inline examples/PBDL/output/trained_model.pth:1]: 1. 这个文件和checkpoint.pth有什么区别，如果是重复的文件，可以删除，否则请转换成`.pdparams`格式的文件

**Comment by HydrogenSulfate** [comment]
@cloud2009 另外请参考：<https://paddlescience-docs.readthedocs.io/zh-cn/latest/zh/development/#3>，补充案例文档

File: examples/PBDL/ibm.py (hunk 1 above)
Comment (cloud2009) [reply examples/PBDL/ibm.py:61]: Thank you! The main(OUTPUT_DIR) function is intended to be used later. I’ll keep it and add at least a pass statement to avoid syntax issues. Implementation is coming soon.

File: examples/PBDL/ibm.py (hunk 2 above)
Comment (cloud2009) [reply examples/PBDL/ibm.py:177]: 这两段 for 循环的作用是提取训练集和测试集中的第一个batch 数据，并将其移动到对应的计算设备（如 GPU 或 CPU）。其中的 break 语句用于确保循环只执行一次，防止遍历完整个数据集。主要用途是用于调试或验证数据加载过程是否正确。如果后续进入正式训练阶段，这两个 break 语句是可以删除的，从而让训练/测试完整遍历整个数据集。

File: examples/PBDL/ibm.py (hunk 3 above)
Comment (cloud2009) [reply examples/PBDL/ibm.py]: 感谢提醒！我会将相关路径拼接方式统一改为 os.path.join(OUTPUT_DIR, ...) 的形式。

File: examples/PBDL/output/TestingLoss_L2.dat
Comment (cloud2009) [reply examples/PBDL/output/TestingLoss_L2.dat:1]: 这是测试过程中实时监测的Loss文件，我们会在后续提交中将它们移除。

File: examples/PBDL/output/TrainingLoss_L2.dat
Comment (cloud2009) [reply examples/PBDL/output/TrainingLoss_L2.dat:1]: 这是训练过程中监测的Loss文件，我们会在后续提交中将它们移除。

File: examples/PBDL/output/checkpoint.pth
Comment (cloud2009) [reply examples/PBDL/output/checkpoint.pth:1]: 好的

File: examples/PBDL/output/loss.dat
Comment (cloud2009) [reply examples/PBDL/output/loss.dat:1]: 好的

File: examples/PBDL/output/nohup.out
Comment (cloud2009) [reply examples/PBDL/output/nohup.out:1]: 好的

File: examples/PBDL/output/trained_model.pth
Comment (cloud2009) [reply examples/PBDL/output/trained_model.pth:1]: 感谢您的提醒！这两个文件的主要区别如下：
1）checkpoint.pth：保存的是训练过程中的中间权重，同时还包含了优化器状态、当前轮次、loss 等信息，方便训练中断后继续训练（resume training）。
2）trained_model.pth：保存了最终的模型参数，用于推理或测试阶段的加载。
